src/java.base/share/classes/sun/security/provider/certpath/RevocationChecker.java

```diff
@@ -740,9 +746,37 @@ private void checkOCSP(X509Certificate cert,
                         null, -1);
                 }
 
+                List<Extension> tmpExtensions = new ArrayList<Extension>();
+                if (rp.ocspNonce) {
+                    if (nonce == null) {
+                        try {
+                            // create the 16-byte nonce by default
+                            Extension nonceExt = new OCSPNonceExtension(DEFAULT_NONCE_BYTES);
+                            tmpExtensions.add(nonceExt);
```

Comment:
I think you should add the OCSPNonce extension to the list of extensions that the application passed in, as there may be other extensions that have been specified and should be sent in the OCSP response, ex:

`ocspExtensions.add(new OCSPNonceExtension(DEFAULT_NONCE_BYTES));`

This means you don't need the `tmpExtensions` variable.

Reply:
During testing, I got the "java.base/java.util.Collections$UnmodifiableCollection.add(Collections.java:1062) exception with this line of change. I've changed to use a tmpExtensions variable when setting the OCSP nonce to the extension sets instead of modifying the ocspExtensions.